integration/tests/access/cohort4/access_test.go

```diff
@@ -106,51 +97,55 @@ func (s *AccessSuite) TestAPIsAvailable() {
 		defer cancel()
 
 		grpcAddress := s.net.ContainerByName(testnet.PrimaryAN).Addr(testnet.GRPCPort)
-		conn, err := grpc.DialContext(ctx, grpcAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		conn, err := grpc.NewClient(grpcAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
 		require.NoError(t, err, "failed to connect to access node")
 		defer conn.Close()
 
 		client := accessproto.NewAccessAPIClient(conn)
 
-		_, err = client.Ping(s.ctx, &accessproto.PingRequest{})
+		_, err = client.Ping(ctx, &accessproto.PingRequest{})
 		assert.NoError(t, err, "failed to ping access node")
 	})
 }
 
-// TestSignerIndicesDecoding tests that access node uses signer indices' decoder to correctly parse encoded data in blocks.
+// runTestSignerIndicesDecoding tests that access node uses signer indices' decoder to correctly parse encoded data in blocks.
 // This test receives blocks from consensus follower and then requests same blocks from access API and checks if returned data
 // matches.
-func (s *AccessSuite) TestSignerIndicesDecoding() {
-
+// CAUTION: must be run last if running multiple tests using the same network since it stops the containers.
+func (s *AccessSuite) runTestSignerIndicesDecoding() {
 	container := s.net.ContainerByName(testnet.PrimaryAN)
 
 	ctx, cancel := context.WithCancel(s.ctx)
 	defer cancel()
 
 	// create access API
 	grpcAddress := container.Addr(testnet.GRPCPort)
-	conn, err := grpc.DialContext(ctx, grpcAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	conn, err := grpc.NewClient(grpcAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	require.NoError(s.T(), err, "failed to connect to access node")
 	defer conn.Close()
 
 	client := accessproto.NewAccessAPIClient(conn)
 
-	// query latest finalized block
-	latestFinalizedBlock, err := MakeApiRequest(client.GetLatestBlockHeader, ctx, &accessproto.GetLatestBlockHeaderRequest{
-		IsSealed: false,
-	})
-	require.NoError(s.T(), err)
-
+	// query latest finalized block. wait until at least two blocks have been finalized.
```

Comment:
this test was passing by accident because it always got the root block. updated to use a real network so blocks were produced, and waiting to check a block that actually had signers